src/views/CreateType.vue

```diff
@@ -0,0 +1,222 @@
+          :autofocus="true"
+          :required="true"
+        />
+        <Select
+          :value.sync="editingAttribute.dataType"
+          name="Data Type"
+          :required="true">
+          <option>string</option>
+          <option>number</option>
+          <option>boolean</option>
+        </Select>
+        <Toggle :value.sync="editingAttribute.required" name="Required"/>
+        <Button text="Save Attribute" :onClick="saveAttribute"/>
+      </Li>
+
+      <Li v-for="attribute in attributeListRight" :key="attribute.id" :options="attribute"/>
+      <Li :options="addButtonOptions"/>
+    </ListSection>
+
+    <Button text="Create Resource Type" :onClick="createResourceType"/>
+  </main>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from 'vue-property-decorator';
+import {
+  ResourceTypes,
+  ResourceType,
+  ResourceTypeAttribute,
+} from '@/apis/rembrandt/rembrandt';
+import Li, { LiOptions } from '@/components/Li.vue';
+import ViewHeader from '@/components/ViewHeader.vue';
+import ListSection from '@/components/ListSection.vue';
+import Input from '@/components/Input.vue';
+import Toggle from '@/components/Toggle.vue';
+import Button from '@/components/Button.vue';
+import Select from '@/components/Select.vue';
+import Utils from '@/utils/Utils';
+
+@Component({
+  components: {
+    ListSection,
+    Li,
+    ViewHeader,
+    Input,
+    Toggle,
+    Button,
+    Select,
+  },
+})
+export default class Types extends Vue {
+  // region public static methods
+  public static emptyResourceType(): ResourceType {
+    return {
+      name: '',
+      parentType: '',
+      abstract: false,
+      attributes: [],
+    };
+  }
+
+  public static emptyResourceTypeAttribute(): ResourceTypeAttribute {
+    return {
+      name: '',
+      dataType: 'string',
+      required: true,
+    };
+  }
+  // endregion
+
+  // region private static methods
+  // endregion
+
+  // region public members
+  public resourceTypes: ResourceType[] = [];
+  public newResourceType: ResourceType;
+  public formState: number = 0;
+  public currentlyEditingAttribute: number = -1;
+
+  public get parentTypeList(): LiOptions[] {
+    return Utils.resourceTypesToList(this.resourceTypes, this.selectParentType);
+  }
+
+  public get parentType(): ResourceType {
+    const parentType = this.resourceTypes.find(
+      (resourceType) => resourceType.id === this.newResourceType.parentType,
+    );
+    return parentType || Types.emptyResourceType();
+  }
+
+  public get parentTypeAttributeList(): LiOptions[] {
+    return Utils.resourceTypeAttributesToList(this.parentType.attributes);
+  }
+
+  public get attributeList(): LiOptions[] {
+    return Utils.resourceTypeAttributesToList(
+      this.newResourceType.attributes,
+      this.editAttribute,
+    );
+  }
+
+  public get attributeListLeft(): LiOptions[] {
+    return this.attributeList.filter(
+      (attribute, index) => index < this.currentlyEditingAttribute,
+    );
+  }
+
```

Comment:
what is attributeListLeft & Right? I hav not seen a list splitted in a left and right part in the creation view.